map/bookmark_manager.cpp

```diff
@@ -1141,6 +1142,53 @@ kml::CompilationType BookmarkManager::GetCompilationType(kml::MarkGroupId id) co
   return compilation->second->GetCategoryData().m_type;
 }
 
+void BookmarkManager::SaveTrackRecording(std::string const & name)
+{
+  CHECK_THREAD_CHECKER(m_threadChecker, ());
+  auto const & tracker = GpsTracker::Instance();
+  CHECK(tracker.IsRecentTrackCollectionInitialized(), ("Track should be initialized"));
+  std::vector<location::GpsTrackInfo> recentTrack = tracker.GetRecentTrackCollection();
+  CHECK(!recentTrack.empty(), ("Recorded track should not be empty"));
+
+  kml::TrackData trackData;
+  std::string trackName = name.empty() ? GenerateTrackRecordingName() : name;
+  kml::SetDefaultStr(trackData.m_name, trackName);
+
+  struct kml::MultiGeometry m_geometry;
+  m_geometry.FromGpsInfoPoints(recentTrack);
+  trackData.m_geometry = std::move(m_geometry);
+
+  struct kml::ColorData colorData;
+  colorData.m_rgba = GenerateTrackRecordingColor().GetRGBA();
+  struct kml::TrackLayer layer;
+  layer.m_color = colorData;
+
+  std::vector<kml::TrackLayer> m_layers;
+  m_layers.emplace_back(layer);
+  trackData.m_layers = std::move(m_layers);
```

Comment:
```suggestion
  trackData.m_layers.emplace_back();
```

Is an empty layer needed?

Reply:
Yes. Ans the layer isnt empty:
`m_layers.emplace_back(layer);`

With the empty `m_layers` the track wouldn't have colour and will not be drawn.

Reply:
Isn't an empty (default) layer created in L1163 and then inserted? The proposed code change does exactly that in one line.

Reply:
Missed colorData somehow, my bad. Thanks!